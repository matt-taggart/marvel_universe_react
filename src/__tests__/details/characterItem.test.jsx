import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CharacterItem from '../../components/details/characterItem';

Enzyme.configure({ adapter: new Adapter() })

describe('Test Character Item', () => {
  const props = {
    name: 'Spiderman',
    resourceURI: '/marvel.com/1234',
  };

  it('Renders correctly', () => {
    const wrapper = shallow(
      <CharacterItem
        name={props.name}
        resourceURI={props.resourceURI}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Link to correct url', () => {
    const wrapper = shallow(
      <CharacterItem
        name={props.name}
        resourceURI={props.resourceURI}
      />,
    );
    expect(wrapper.find('Link').prop('to')).toEqual('/characters/1234');
  });
});
