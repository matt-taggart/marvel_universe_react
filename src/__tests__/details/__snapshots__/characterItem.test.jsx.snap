// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Item Should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CharacterItem
    name="Spiderman"
    resourceURI="/marvel.com/1234"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="level-left"
      >
        <div
          className="level-item center-mobile"
        >
          Spiderman
        </div>
        <small
          className="level-item"
        >
          <Link
            replace={false}
            to="/characters/1234"
          >
            Details
          </Link>
        </small>
      </div>,
      "className": "level",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="level-item center-mobile"
          >
            Spiderman
          </div>,
          <small
            className="level-item"
          >
            <Link
              replace={false}
              to="/characters/1234"
            >
              Details
            </Link>
          </small>,
        ],
        "className": "level-left",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Spiderman",
            "className": "level-item center-mobile",
          },
          "ref": null,
          "rendered": "Spiderman",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/characters/1234"
            >
              Details
            </Link>,
            "className": "level-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Details",
              "replace": false,
              "to": "/characters/1234",
            },
            "ref": null,
            "rendered": "Details",
            "type": [Function],
          },
          "type": "small",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="level-left"
        >
          <div
            className="level-item center-mobile"
          >
            Spiderman
          </div>
          <small
            className="level-item"
          >
            <Link
              replace={false}
              to="/characters/1234"
            >
              Details
            </Link>
          </small>
        </div>,
        "className": "level",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="level-item center-mobile"
            >
              Spiderman
            </div>,
            <small
              className="level-item"
            >
              <Link
                replace={false}
                to="/characters/1234"
              >
                Details
              </Link>
            </small>,
          ],
          "className": "level-left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Spiderman",
              "className": "level-item center-mobile",
            },
            "ref": null,
            "rendered": "Spiderman",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/characters/1234"
              >
                Details
              </Link>,
              "className": "level-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Details",
                "replace": false,
                "to": "/characters/1234",
              },
              "ref": null,
              "rendered": "Details",
              "type": [Function],
            },
            "type": "small",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
